src/com/owncloud/android/utils/ErrorMessageAdapter.java

```diff
@@ -173,7 +173,11 @@ public static String getErrorCauseMessage(RemoteOperationResult result,
             }
         } else if (operation instanceof CreateShareViaLinkOperation ||
                     operation instanceof CreateShareWithShareeOperation) {
-            if (result.getCode() == ResultCode.SHARE_NOT_FOUND)  {      // Error --> SHARE_NOT_FOUND
+
```

Comment:
I wouldn't go with the collection. Those mappings are more difficult to read & maintain, and error messages may change quite easily.

About the other idea, I think I see your point. The messages about permissions need an overall review. But trying to refactor the messages around around their structure is dangerous because of the translations to languages with very different syntax. It's a topic we can revisit in a while, since we'll have to extend this dialog soon with the options to change permissions on the shares.